wrappers/services.go

```diff
@@ -375,120 +376,202 @@ func userDaemonReload() error {
 	return cli.ServicesDaemonReload(ctx)
 }
 
-// AddSnapServicesOptions is a struct for controlling the generated service
-// definition for a snap service.
-type AddSnapServicesOptions struct {
-	Preseeding              bool
-	VitalityRank            int
-	RequireMountedSnapdSnap bool
-}
-
-// AddSnapServices adds service units for the applications from the snap which
-// are services. The services do not get enabled or started.
-func AddSnapServices(s *snap.Info, opts *AddSnapServicesOptions, inter interacter) (err error) {
-	if s.Type() == snap.TypeSnapd {
-		return fmt.Errorf("internal error: adding explicit services for snapd snap is unexpected")
-	}
+// EnsureSnapServices will ensure that the specified snap services' file states
+// are up to date with the specified options and infos.
```

Comment:
Maybe adding info here that this will not handle changed services, i.e. if a service is added/removed this will not be reflected? Not sure what the longer term thinking here is, if that should be handled eventually.

Reply:
it should work for added services otherwise the name is strange?  for removing I'm not sure it's interesting to support it here

Reply:
Well, this does handle added services properly, which was not necessarily the intention, but a natural consequence of the implementation, so I switched AddSnapServices to use this since it reduces the code duplication and complexity of that function. 

The difficulty of implementing a version that removes services that are no longer present in the snap's `Apps` list is that we need to track the existing state, which would require either:

* passing into this function the old state of the snap's `SnapInfo` which would be awkward
* somehow grepping or inspecting systemctl output to find all the existing snap services for a given snap, and then comparing that to the set of services in the snap currently, which seems brittle and error prone

I think given that this is meant primarily to be used when we are not refreshing the app snaps in question, but changing snap's service's unit definitions (i.e. it's resource groups or its inter snap dependencies or maybe some snapd config for services like the docker CVE we had recently with `Delegates`), I don't think it's worth the complexity to try and implement handling removed services.

I will add something to the doc-comment summarizing the fact that:
* adding new snap services works properly
* removing snap services is not handled